lib/generators/settings_model/settings_model_generator.rb

```diff
@@ -0,0 +1,59 @@
+# Based on https://github.com/huacnlee/rails-settings-cached/blob/main/lib/generators/settings/install_generator.rb
+
+require "rails/generators"
+require "rails/generators/migration"
+
+class SettingsModelGenerator < Rails::Generators::NamedBase
+  include Rails::Generators::Migration
+
+  argument :name, type: :string, default: "setting"
+
+  source_root File.expand_path("templates", __dir__)
+
+  @migrations = false
+
+  def self.next_migration_number(dirname) #:nodoc:
+    if ActiveRecord::Base.timestamped_migrations
+      if @migrations
+        (current_migration_number(dirname) + 1)
+      else
+        @migrations = true
+        Time.now.utc.strftime("%Y%m%d%H%M%S")
+      end
+    else
+      format "%.3<number>d", number: current_migration_number(dirname) + 1
+    end
+  end
+
+  def create_migration_file
+    migration_template(
+      "migration.erb",
+      "db/migrate/create_settings.rb",
+      migration_version: migration_version,
+      table_name: table_name,
+    )
+  end
+
+  def create_model
+    template(
+      "model.erb",
+      File.join("app/models/settings", class_path, "#{file_name}.rb"),
+    )
+  end
+
+  def rails_version_major
+    Rails::VERSION::MAJOR
+  end
+
+  def rails_version_minor
+    Rails::VERSION::MINOR
+  end
+
+  def migration_version
+    "[#{rails_version_major}.#{rails_version_minor}]" if rails_version_major >= 5
+  end
+
+  def table_name
+    @table_name ||= ":settings_#{class_name.underscore.pluralize}"
```

Comment:
This is my only real gripe: 

1. I'd love for the tables to all have a `settings_` prefix, because it nicely groups them together.
2. But that means we'll end up with table names like `settings_notifications`, which are a bit confusing.

It's probably better to switch it to `notification_settings`, WDYT?

Reply:
I actually like the tables to be grouped together as `settings_something`

Reply:
The problem is that it would need to be pluralized due to Rails conventions and some names become potentially confusing, ae.g. `settings_authentications` vs. `authentication_settings`. The former sort of suggests that it stores the authentications for accessing settings or something.

Reply:
I also like prefix grouping but I agree `settings_notifications` is a bit :dizzy_face: . That'll just have to be an acceptable trade-off. TBF adding `_notifications` behind anything adds confusion.